code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -17,8 +18,16 @@ import (
 
 var ErrInvalidEventData = errors.New("invalid event data")
 
-func (edb *EventDb) ProcessEvents(ctx context.Context, events []Event, round int64, block string, blockSize int) error {
+func (edb *EventDb) ProcessEvents(
+	ctx context.Context,
+	events []Event,
+	round int64,
+	block string,
+	blockSize int,
+	cfg config.DbAccess,
+) error {
 	ts := time.Now()
+	edb.updateSettings(cfg)
```

Comment:
We'd better not update config every time when the ProcessEvents() is called, this function will be called frequently,  so we should update it when necessary, e.g config is changed or initialized, etc.

Reply:
You mean to add a dirty field to the chain.ChainConfig to indicate it changed this round.